.github/workflows/validate-examples.yml

```diff
@@ -0,0 +1,28 @@
+name: Validate AsyncAPI Examples
+
+on:
+  push:
+    branches:
+      - script-ci-spec-validation # added temporarily for testing purpose
+      # - master
+
+  pull_request_target:
+    types: [opened, reopened, synchronize, edited, ready_for_review]
+
+jobs:
+  validate-examples:
+    name: Validate Spec Examples
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+      - name: Install AsyncAPI CLI
+        run: npm install -g @asyncapi/cli
+      - name: Validate AsyncAPI documents
+        run: |
+          find examples -type f \( -name "*.yml" -o -name "*.yaml" \) -not -path 'examples/social-media/common/*' | xargs -P 10 -L 1 asyncapi validate 
```

Comment:
omg, bash on steroids

luckily we have chat gpt to translate it for me when needed 😁

Reply:
> omg, bash on steroids
> 
> luckily we have chat gpt to translate it for me when needed 😁

:laughing:

Reply:
Come on! are self explanatory 😆 Let me save your time

```sh
find examples -type f \( -name "*.yml" -o -name "*.yaml" \) -not -path 'examples/social-media/common/*' 
```

This is completely human readable: find in examples dir "things" of type file (f) with name suffixes `yml` or `yaml` and do not include the path  'examples/social-media/common/*' . The output is a list of paths to those files.

```sh
xargs -P 10 -L 1 asyncapi validate 
```

Here maybe the trickiest part: xargs meaning do "x" with the piped arguments (which is the list of files), the -L 1 means take each new line as one execution, -P 10 means parallellize in 10 runs at the same time, and for each of those arguments (paths), execute `asyncapi validate` command.